library/dd-sdk-android-session-replay-material/src/main/kotlin/com/datadog/android/sessionreplay/material/SliderWireframeMapper.kt

```diff
@@ -0,0 +1,153 @@
+
+internal open class SliderWireframeMapper(
+    private val viewUtils: ViewUtils = ViewUtils,
+    private val stringUtils: StringUtils = StringUtils,
+    private val uniqueIdentifierGenerator: UniqueIdentifierGenerator =
+        UniqueIdentifierGenerator
+) :
+    WireframeMapper<Slider, MobileSegment.Wireframe> {
+
+    @Suppress("LongMethod")
+    override fun map(view: Slider, systemInformation: SystemInformation):
+        List<MobileSegment.Wireframe> {
+        val activeTrackId = uniqueIdentifierGenerator
+            .resolveChildUniqueIdentifier(view, TRACK_ACTIVE_KEY_NAME)
+        val nonActiveTrackId = uniqueIdentifierGenerator
+            .resolveChildUniqueIdentifier(view, TRACK_NON_ACTIVE_KEY_NAME)
+        val thumbId = uniqueIdentifierGenerator.resolveChildUniqueIdentifier(view, THUMB_KEY_NAME)
+
+        if (activeTrackId == null || thumbId == null || nonActiveTrackId == null) {
+            return emptyList()
+        }
+
+        val screenDensity = systemInformation.screenDensity
+        val viewGlobalBounds = viewUtils.resolveViewGlobalBounds(view, screenDensity)
+        val normalizedSliderValue = view.normalizedValue()
+        val viewAlpha = view.alpha
+
+        // padding
+        val trackLeftPadding = view.trackLeftPadding(screenDensity)
+        val trackTopPadding = view.paddingTop.toLong().densityNormalized(screenDensity)
+
+        // colors
+        val drawableState = view.drawableState
+        val trackActiveColor = view.trackActiveTintList.getColor(drawableState)
+        val trackNonActiveColor = view.trackInactiveTintList.getColor(drawableState)
+        val thumbColor = view.thumbTintList.getColor(drawableState)
+        val trackActiveColorAsHexa = stringUtils.formatColorAndAlphaAsHexa(
+            trackActiveColor,
+            OPAQUE_ALPHA_VALUE
+        )
+        val trackNonActiveColorAsHexa = stringUtils.formatColorAndAlphaAsHexa(
+            trackNonActiveColor,
+            PARTIALLY_OPAQUE_ALPHA_VALUE
+        )
+        val thumbColorAsHexa = stringUtils.formatColorAndAlphaAsHexa(thumbColor, OPAQUE_ALPHA_VALUE)
+
+        // track dimensions
+        val trackWidth = view.trackWidth.toLong().densityNormalized(screenDensity)
+        val trackHeight = view.trackHeight.toLong().densityNormalized(screenDensity)
+        val trackActiveWidth = (trackWidth * normalizedSliderValue).toLong()
+
+        // track positions
+        val trackXPos = viewGlobalBounds.x + trackLeftPadding
+        val trackYPos = viewGlobalBounds.y + trackTopPadding +
+            (viewGlobalBounds.height - trackHeight) / 2
+
+        // thumb dimensions
+        val thumbHeight = (view.thumbRadius.toLong().densityNormalized(screenDensity)) * 2
+
+        // thumb positions
+        val thumbXPos = (trackXPos + trackWidth * normalizedSliderValue).toLong()
+        val thumbYPos = viewGlobalBounds.y + trackTopPadding +
+            (viewGlobalBounds.height - thumbHeight) / 2
```

Comment:
will it work for vertical slider as well?

Reply:
good point...and that's a SeekBar for which I will have another PR.